color_color.py

```diff
@@ -0,0 +1,738 @@
+#!/usr/bin/env python
+
+#
+# Copyright (C) 2017, 2019 Smithsonian Astrophysical Observatory
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+
+"""Color Color Plot maker
+
+Creates a plot showing how model parameters affect hardness ratios
+given a user supplied ARF and RMF.
+
+"""
+
+import numpy as np
+import sherpa.astro.ui as ui
+
+# import sys must follow sherpa (sorry pylint)
+import sys
+sys.tracebacklimit = 999
+
+
+__all__ = ["EnergyBand", "ModelParameter", "HardnessRatioAxis",
+           "ColorColor", "ColorColorDiagram"]
+
+
+def make_acis_diagonal_rmf(arf):
```

Comment:
You should be able to use `sherpa.astro.instrument.crete_delta_rmf` here. If not then this suggests problems/improvements are on the cards for `create_delta_rmf`...

Reply:
Hmm. Maybe not, as `create_delta_rmf` assumes a one-to-one mapping between channel and energy bin and here we have 1024 channels. It would be nice to add to `create_delta_rmf`, but that's a different game.